storm-core/src/jvm/backtype/storm/serialization/ThriftSerializationDelegate.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package backtype.storm.serialization;
+
+import org.apache.thrift.TBase;
+import org.apache.thrift.TDeserializer;
+import org.apache.thrift.TException;
+import org.apache.thrift.TSerializer;
+
+import java.util.Map;
+
+public class ThriftSerializationDelegate implements SerializationDelegate {
```

Comment:
I don't believe that this will ever work Standalone.  I don't see how we could ever make a custom grouping that is a subclasses of TBase, and even if we could the code in ThriftSerializationDelegateBridge would break, because we would write it out using thrift, but try to read it back in using java serialization.  I am not really sure the best way to handle this though.  The only real thing that comes to mind would be to have different serialization delegates for different places we store data.  One would be for the topology instances, which we could leave using java serialization because it is part of the contract we made in the API for bolts, spouts, and groupings. The other would be for everything else.